Dockerfile

```diff
@@ -0,0 +1,44 @@
+# syntax=docker/dockerfile:1.4
+
+# Use Ubuntu 22.04 as the build environment
+FROM ubuntu:22.04 as build-environment
+
+# Set non-interactive mode for apt so it doesn't ask for user input during the build
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install required dependencies
+RUN apt-get update && apt-get install -y clang lld curl build-essential linux-generic git \
+    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
+    && chmod +x ./rustup.sh \
+    && ./rustup.sh -y
+    
+ARG TARGETARCH
+WORKDIR /opt
+
+
+# Add the required CFLAGS if the TARGETARCH matches
+RUN [[ "$TARGETARCH" = "x86_64-unknown-linux-gnu" ]] && echo "export CFLAGS=-mno-outline-atomics" >> $HOME/.profile || true
+
+WORKDIR /opt/cryo
+COPY . .
+
+RUN . $HOME/.profile && cargo build --bin cryo --release \
```

Comment:
it would be good to use `--locked` flag to avoid unexpected dependency updates during docker build, which should in theory just translate repo as is to a built form

Reply:
done